packages/clerk-js/src/ui/components/UserButton/SessionActions.tsx

```diff
@@ -1,20 +1,53 @@
 import type { ActiveSessionResource } from '@clerk/types';
 
 import type { ElementDescriptor, ElementId } from '../../../ui/customizables/elementDescriptors';
+import { useRouter } from '../../../ui/router';
+import { USER_BUTTON_ITEM_ID } from '../../constants';
+import { useUserButtonContext } from '../../contexts';
 import type { LocalizationKey } from '../../customizables';
 import { descriptors, Flex, localizationKeys } from '../../customizables';
 import { Action, Actions, PreviewButton, SmallAction, SmallActions, UserPreview } from '../../elements';
 import { Add, CogFilled, SignOut, SwitchArrowRight } from '../../icons';
 import type { ThemableCssProp } from '../../styledSystem';
+import type { DefaultItemIds, MenuItem } from '../../utils/createCustomMenuItems';
 
 type SingleSessionActionsProps = {
   handleManageAccountClicked: () => Promise<unknown> | void;
   handleSignOutSessionClicked: (session: ActiveSessionResource) => () => Promise<unknown> | void;
+  handleUserProfileActionClicked: (startPath?: string) => Promise<unknown> | void;
   session: ActiveSessionResource;
+  completedCallback: () => void;
 };
 
 export const SingleSessionActions = (props: SingleSessionActionsProps) => {
-  const { handleManageAccountClicked, handleSignOutSessionClicked, session } = props;
+  const { navigate } = useRouter();
+  const { handleManageAccountClicked, handleSignOutSessionClicked, handleUserProfileActionClicked, session } = props;
+
+  const { menutItems } = useUserButtonContext();
+
+  const commonActionSx: ThemableCssProp = t => ({
+    borderTopWidth: t.borderWidths.$normal,
+    borderTopStyle: t.borderStyles.$solid,
+    borderTopColor: t.colors.$neutralAlpha100,
+    padding: `${t.space.$4} ${t.space.$5}`,
+  });
+
+  const handleActionClick = async (route: MenuItem) => {
```

Comment:
I saw `route.onClick` below, and it was just very confusing
```suggestion
  const handleActionClick = async (menuItem: MenuItem) => {
```

Reply:
Done